keps/sig-apps/3998-job-success-completion-policy/README.md

```diff
@@ -0,0 +1,924 @@
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, we can disable the `JobSuccessPolicy` feature gate.
+When the feature is disabled, the job controller stop evaluating the successPolicy even if
+the `.spec.successPolicy` is set.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The Job controller considers the `.spec.successPolicy` when it updates `.status.conditions`
+only for running Jobs and don't update `.status.conditions` for already finished jobs.
+
+###### Are there any tests for feature enablement/disablement?
+
+No, we'll add them during alpha.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Even if the kube-controller-manager is rolled out or rollback fail, already running workloads aren't any impact.
+The default behavior will be applied to running workloads.
+
+###### What specific metrics should inform a rollback?
+
+An increase in the `job_sync_duration_seconds` metrics can mean that finished jobs
+are taking longer to evaluate.
+
+The users should check whether the completed jobs have the appropriate condition,
+specifically the reason.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+No.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+`job_finished_total`: the new `JobSuccessPolicy` and `Completions` values for the `reason` label. 
+  - `JobSuccessPolicy` indicates a job is declared as succeeded because the job meets `.spec.successPolicy`.
+  - `Completions` indicates a job is declared as succeeded because the job meets `spec.completions`.
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [x] Job API .status
+  - The Job controller will add a condition with `JobSuccessPolicy` reason to `conditions`.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+99% percentile over day for Job syncs is <= 15s for a client-side 50 QPS limit.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [x] Metrics
+  - Metric name: `job_sync_duration_seconds` (existing): can be used to see how much the
+    feature enablement increases the time spent in the sync job
+  - Components exposing the metric:
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+No.
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No.
+
+### Scalability
+
+No.
+
+###### Will enabling / using this feature result in any new API calls?
+
+No.
```

Comment:
We can probably have additional patch to set the new status if we face the condition (which is absolutely ok, but maybe worth mentioning).

Reply:
Good point :)
Let me mention it here.

Reply:
Done.